### ISSUE TITLE
Respect `window_gap` when running `window --grid` commands?

### ISSUE THREAD

**Issue by jesseleite** [koekeishiya/yabai#932]
Firstly let me say that I'm loving yabai; Thank you for creating such a great tool! ❤️

Anyway, in my previous hammerspoon-based floating-window grid-layout manager, I had lots of padding around my windows...

![CleanShot 2021-06-19 at 13 02 47](https://user-images.githubusercontent.com/5187394/122650167-59659700-d0ff-11eb-830e-a282db437823.png)

Yabai's padding settings work perfectly when in a bsp layout...

```
yabai -m config layout                       bsp
yabai -m config top_padding                  36
yabai -m config bottom_padding               36
yabai -m config left_padding                 36
yabai -m config right_padding                36
yabai -m config window_gap                   36
```

However when in a float layout, it seems yabai only respects the outer padding, but not my `window_gap`. For example, if I run these `yabai -m window --grid` commands on the same two windows, I get this...

![CleanShot 2021-06-19 at 13 03 59](https://user-images.githubusercontent.com/5187394/122650213-a21d5000-d0ff-11eb-8695-cfada60bf5ec.png)

Thoughts on how this could be implemented...

- If the edge of the grid cell is against the edge of the screen, use the configured padding (as you already do).
- If the edge of the grid cell is not against the edge of the screen, apply half of the `window_gap`, so that any adjacent grid cells will show the configured `window_gap`.
    - Maybe this could require a `--with-window-gap` for backwards compatibility, if that's a concern?
    
What are your thoughts?

**Comment by koekeishiya** [comment]
I don't think there needs to be a setting for this, should probably be default behaviour. 
Those who do not want this functionality should just set the window gap to zero in floating spaces, or just toggle `yabai -m space --toggle gap` before doing grid commands.

**Comment by jesseleite** [comment]
Would be loverly if it was default behaviour! ❤️

**Comment by koekeishiya** [comment]
Should be fixed on master.

**Comment by koekeishiya** [comment]
Fixed in v7.0.0.
